channels/channel.nos/nos2010/chn_nos2010.py

```diff
@@ -852,27 +852,42 @@ def extract_api_pages(self, data):
         next_url = data.get_value("_links", "next", "href")
         Logger.debug("Retrieving a total of %s items", data.get_value("total"))
 
-        if not next_url:
-            return data, items
-
-        # We will just try to download all items.
-        for _ in range(0, self.__max_page_count - 1):
-            page_data = UriHandler.open(next_url, additional_headers=self.parentItem.HttpHeaders)
-            page_json = JsonHelper(page_data)
-            page_items = page_json.get_value("items")
-            if page_items:
-                data.json["items"] += page_items
-            next_url = page_json.get_value("_links", "next", "href")
-            if not next_url:
-                break
-
         if next_url:
-            next_title = LanguageHelper.get_localized_string(LanguageHelper.MorePages)
-            item = FolderItem("\b.: {} :.".format(next_title), next_url, content_type=contenttype.EPISODES)
-            item.complete = True
-            item.HttpHeaders = self.__jsonApiKeyHeader
-            item.dontGroup = True
-            items.append(item)
+            # We will just try to download all items.
+            for _ in range(0, self.__max_page_count - 1):
+                page_data = UriHandler.open(next_url, additional_headers=self.parentItem.HttpHeaders)
+                page_json = JsonHelper(page_data)
+                page_items = page_json.get_value("items")
+                if page_items:
+                    data.json["items"] += page_items
+                next_url = page_json.get_value("_links", "next", "href")
+                if not next_url:
+                    break
+
+            if next_url:
+                next_title = LanguageHelper.get_localized_string(LanguageHelper.MorePages)
+                item = FolderItem("\b.: {} :.".format(next_title), next_url, content_type=contenttype.EPISODES)
+                item.complete = True
+                item.HttpHeaders = self.__jsonApiKeyHeader
+                item.dontGroup = True
+                items.append(item)
+
+        # Check if there are extras for this show.
+        first_url = data.get_value("_links", "first", "href")
+        if re.match(r'https://start-api.npo.nl/media/series/[^/]+/episodes', first_url):
+            extras_url = first_url.replace("/episodes", "/clips")
+            Logger.debug("Looking for extras: %s", extras_url)
+            extras_data = UriHandler.open(extras_url, additional_headers=self.parentItem.HttpHeaders)
+            extras_json = JsonHelper(extras_data)
+            extras_items = extras_json.get_value("items")
+
+            if len(extras_items) > 0:
+                Logger.debug("Adding link to extras: %s", extras_url)
+                item = FolderItem("\a.: Extra :.", extras_url, content_type=contenttype.EPISODES)
```

Comment:
Would it make sense to call it "Clips" instead of "Extra's"? Then you can reuse the existing translation for that:

```python
clips = LanguageHelper.get_localized_string(LanguageHelper.Clips)
clip_title = "\a.: %s :." % (clips,)
```

Or should we make the "Extra's" a translation?

Reply:
I think that "Extra's" would be better, because that's what they are called on the website and in the app. "Clips" could also mean "Fragmenten".

Reply:
Ok, then let's add "Extras" to the `resources/language/resource.language.en_gb/strings.po` and perhaps translate it in `resources/language/resource.language.nl_nl/strings.po`. This can be with ID 30368: https://github.com/retrospect-addon/plugin.video.retrospect/blob/master/resources/language/resource.language.en_gb/strings.po#L678

Then add `Extras` to https://github.com/retrospect-addon/plugin.video.retrospect/blob/master/resources/lib/helpers/languagehelper.py#L74 and assign it 30368. That way you can do:

```python 
extras = LanguageHelper.get_localized_string(LanguageHelper.Extras)
extras_title = "\a.: %s :." % (extras,)
```

Reply:
OK.